streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignorTest.java

```diff
@@ -632,6 +633,106 @@ public void testAssignWithInternalTopicThatsSourceIsAnotherInternalTopic() throw
         assertEquals(allTasks.size(), (long) internalTopicManager.readyTopics.get("test-topicZ"));
     }
 
+    @Test
+    public void shouldGenerateTasksForAllCreatedPartitions() throws Exception {
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final InternalTopologyBuilder internalTopologyBuilder = StreamsBuilderTest.internalTopologyBuilder(builder);
+        internalTopologyBuilder.setApplicationId(applicationId);
+
+        // KStream with 3 partitions
+        KStream<Object, Object> stream1 = builder
+            .stream("topic1")
+            // force creation of internal repartition topic
+            .map(new KeyValueMapper<Object, Object, KeyValue<Object, Object>>() {
+                @Override
+                public KeyValue<Object, Object> apply(final Object key, final Object value) {
+                    return new KeyValue<>(key, value);
+                }
+            });
+
+        // KTable with 4 partitions
+        KTable<Object, Long> table1 = builder
+            .table("topic3")
+            // force creation of internal repartition topic
+            .groupBy(new KeyValueMapper<Object, Object, KeyValue<Object, Object>>() {
+                @Override
+                public KeyValue<Object, Object> apply(final Object key, final Object value) {
+                    return new KeyValue<>(key, value);
+                }
+            })
+            .count(Materialized.<Object, Long, KeyValueStore<Bytes, byte[]>>as("count"));
```

Comment:
We actually don't need to name the store. This could be `.count()` plus updating the name for the repartitioning and changelog topic.

Reply:
know the naming was not required. My thinking is it would be preferable to explicitly choose the name so I don't have to rely on internal logic for naming the repartition topic that I later refer to. Unfortunately there's no explicit naming for the streams.map() repartition topic so it is all moot anyway.

Reply:
I understand your point -- however, naming must be deterministic and should not change from release to release (otherwise, upgrading hard harder and we need to mention it in the upgrade docs). Thus, if we use internal names in tests, we have some implicit testing that naming does not change. That's why I prefer to not name the operator, too, if not required.

Reply:
Yes I guess that's ok. I agree that an upgrade should not require a reset of the streams application's topology unless absolutely necessary. However I think in that case we should also have some explicit testing for the naming of itnernal topics (not sure if those exist yet?)

Either way I will fix this PR up once I am back in the office.